torch/nested/_internal/nested_tensor.py

```diff
@@ -13,7 +12,14 @@
 
 
 def get_tensor_symint(tensor, *, coeff=1):
+    from torch._subclasses.fake_tensor import FakeTensor
+    from torch._subclasses.functional_tensor import FunctionalTensor
+
+    if isinstance(tensor, (FakeTensor, FunctionalTensor)):
+        return tensor.get_nested_int(coeff=coeff)
```

Comment:
confirming my understanding, but:

(1) one invariant we want is that the global `_tensor_symint_registry` is never mutated during tracing (and only contains real tracing)
(2) that invariant is maintained here via the short-circuit
(3) we get "reasonable" behavior today because at runtime, any offset tensors that become inputs to NJT constructors in the graph (that were previously not associated with an NJT) will get added to the dict here in AOTDispatcher's runtime wrapper on exit (when we wrap NJT tensor subclass graph outputs)

Reply:
Yep, that is correct